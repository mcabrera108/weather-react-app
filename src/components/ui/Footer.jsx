import style from "../../style_modules/layout.module.css";
function Footer() {
  return (
    <footer className={style.footerContainer}>
      <div className={style.footerWrapper}>
        <h4>@ 2024-2025 Martin Cabrera. All rights reserved.</h4>
      </div>
    </footer>
  );
}
export default Footer;
