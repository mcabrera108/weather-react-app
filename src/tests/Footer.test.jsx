import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Footer from "../components/ui/Footer";

describe("Footer Component", () => {
  it("Renders Footer", () => {
    const container = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it("Renders correct text inside Footer", () => {
    const container = render(<Footer />);
    expect(container.getByRole("heading").textContent).toMatch(
      "@ 2024-2025 Martin Cabrera. All rights reserved."
    );
  });
});
