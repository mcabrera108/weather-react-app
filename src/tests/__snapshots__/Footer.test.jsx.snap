// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > Renders Footer 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="_footerContainer_4f5d64"
      >
        <div
          class="_footerWrapper_4f5d64"
        >
          <h4>
            @ 2024-2025 Martin Cabrera. All rights reserved.
          </h4>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="_footerContainer_4f5d64"
    >
      <div
        class="_footerWrapper_4f5d64"
      >
        <h4>
          @ 2024-2025 Martin Cabrera. All rights reserved.
        </h4>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
